java/ql/lib/semmle/code/java/dataflow/internal/DataFlowUtil.qll

```diff
@@ -304,6 +304,27 @@ class ContentSet instanceof Content {
   }
 }
 
+/**
+ * Holds if `g` validates the `e` upon evaluating to `branch`.
+ *
+ * The expression `e` is expected to be a syntactic part of the guard `g`.
+ * For example, the guard `g` might be a call `isSafe(x)` and the expression `e`
+ * the argument `x`.
+ */
+signature predicate guardChecksSig(Guard g, Expr e, boolean branch);
+
+module BarrierGuard<guardChecksSig/3 guardChecks> {
+  /** Gets a node that is safely guarded by the given guard. */
+  Node getABarrierNode() {
```

Comment:
Maybe rename to getAGuardedNode for consistency (looks like this is the higher order version of the getAGuardedNode in the class BarrierGuard.

Reply:
I like `getABarrierNode()` better, since it's supposed to plug directly into `isBarrier(Node)`, so I think this will be more intuitive. Also, consistency will be trivially achieved once the BarrierGuard class is deleted.